migration/migration.go

```diff
@@ -183,6 +184,10 @@ func streamThroughTempFile(r io.Reader) (_ io.ReadSeeker, cleanup func(), err er
 }
 
 func uploadCharms(config UploadBinariesConfig) error {
+	// It is critical that charms are uploaded in ascending charm URL
+	// order due to the way local charm URLs are tracked.
+	utils.SortStringsNaturally(config.Charms)
```

Comment:
Why is this? (Reads ahead.) 
I think it's so if there are multiple revisions of the same charm they're uploaded in revision order, so that the revisions are the same in the source and target? It'd be good to have that detail in the comment.